application-workloads/darktrace-vsensor-standalone/azuredeploy.json

```diff
@@ -77,6 +77,7 @@
             }
         },
         "adminPublicKey": {
+            "defaultValue": "",
```

Comment:
if this is indeed optional, then we need to remove the param from the param file for the sample so the default is tested...

Reply:
thanks, that's now done

Reply:
Looks like (according to CI) something still isn't right... failed on the missing key.  Is there another property that needs to be set?  otherwise the template is going to send an empty string for the value.

Reply:
azuredeploy.json works - I tested it in 'Template deployment (deploy using custom templates) '. 

All three scenarios worked:
-  Generate new key pair;
- Use existing key stored in Azure;
- Use existing public key;

The only error I could see was with 'Use existing key stored in Azure'  and that could be caused by the empty string to the value:       

![image](https://user-images.githubusercontent.com/98969094/194110460-9e731e88-7893-438f-b72a-ba6f3f687b64.png)


Once a key is selected it works:         

![image](https://user-images.githubusercontent.com/98969094/194110536-959ef2de-29f3-4bd0-82fc-fca505f15128.png)


The template **before** the change works the same way  (same error) with 'Use existing key stored in Azure' and it did not fail the CI.

So, it looks like that if `"defaultValue": ""` in 'new' azuredeploy.json causes this (which probably the case is):

```
    "adminPublicKey": {
      "defaultValue": "",
...
      }
```

then `"value": "GEN-SSH-PUB-KEY"` in the 'old' azuredeploy.parameters.json does the same.

The only thing I can think of is to keep `"defaultValue": ""` in 'new' azuredeploy.json, and to revert azuredeploy.parameters.json back to have ` "adminPublicKey": {"value": "GEN-SSH-PUB-KEY"}` so the pipe line won't fail.

Any other suggestions?

Reply:
A couple things:
1) portal doesn't use the parameter files when deploying (unless you manually submit them) - the param files in the repo are only used for testing, they are not used by any other "client"
2) if something isn't working correctly with the portal's handling of the parameter in the deployment experience (and the template works correctly here) then it's a portal issue - they've had trouble with trying to craft and experience around SSH keys in the past